MinecraftClient/McTcpClient.cs

```diff
@@ -1615,7 +1615,20 @@ public void OnUpdateHealth(float health, int food)
                 }
             }
             foreach (ChatBot bot in bots.ToArray())
-                bot.OnHealthUpdate(health, food);
+            {
+                try
+                {
+                    bot.OnHealthUpdate(health, food);
+                }
+                catch (Exception e)
+                {
+                    if (!(e is ThreadAbortException))
+                    {
+                        ConsoleIO.WriteLogLine("OnPlayerProperty: Got error from " + bot.ToString() + ": " + e.ToString());

```

Comment:
Thanks for adding an error handler here. I thing the message should be `OnHealthUpdate: Got error from [...]`.